File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
  const obj = {};
  
  // 참가자 카운트
  for (let p of participant) {
    obj[p] = (obj[p] || 0) + 1;
  }
  
  // 완주자 카운트 감소
  for (let c of completion) {
    obj[c] -= 1;
  }
  
  // 값이 1인 사람 = 완주 못한 사람
  for (let key in obj) {
    if (obj[key] > 0) return key;
  }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(p, c):
    dic = {}
    for name in p:
        if dic.get(name): dic[name] += 1
        else: dic[name] = 1
    for name in c:
        dic[name] -= 1
    for key in dic:
        if dic[key]: return key
    